hayeon/programmers-2/귤-고르기.js

```diff
@@ -0,0 +1,28 @@
+function solution(k, tangerine) {
+  const obj = {};
+
+  tangerine.forEach((el) => {
+    obj[el] = ++obj[el] || 1;
+  });
+
+  const values = Object.values(obj).sort((a, b) => b - a);
+
+  let count = 0;
+  for (let i = 0; i < values.length; i++) {
+    count += values[i];
+    if (count >= k) {
+      return i ? i + 1 : 1;
+    }
+  }
+}
```

Comment:
```suggestion
function solution(k, tangerine) {
  const tangerineCountObj = {};

  tangerine.forEach((el) => {
    tangerineCountObj[el] = ++tangerineCountObj[el] || 1;
  });

  const tangerineCountes = Object.values(tangerineCountObj).sort((a, b) => b - a);
  let sum = 0;
  
  for (let i = 0; i < tangerineCountes.length; i++) {
    sum += tangerineCountes[i];
    
    if (sum >= k) {
      return i ? i + 1 : 1;
    }
  }
}
```
- 풀이 잘봣습니다
- 변수명 더 신경써주신다고 말씀하셨는데.. 🥲 변수명 좀 더 디테일하게 잡아주시면 좋을 것 같습니다
- 문단 별로 잘 분리해주세요~
- 변수 선언부 중간에 forEach쓰거나, Object.values와 같은 추가 작업 필요없이 해당 문제는 배열로 작성하면 좀 더 깔끔한 구조로 가져갈 수 있습니다.
```js
function solution(k, tangerine) {
  const tangerineCountArr = tangerine.reduce(
    (acc, cur) => {
      acc[cur] = (acc[cur] || 0) + 1;
      return acc;
    },
    [0]
  );
  let sum = 0;
  let result = 0;

  tangerineCountArr.sort((a, b) => b - a);

  for (const count of tangerineCountArr) {
    sum += count;
    result++;

    if (sum >= k) return result;
  }
}
```

Reply:
- 변수명 더 신경쓰겠습니다ㅠㅠ 풀이하는데만 신경쓰느라 간과했네요...
- 피드백 감사합니다 참고해서 풀이 다시 검토해보겠습니다!!🙇‍♀️